### PR TITLE
Maintenance updates for support + CI

### LINKED ISSUE #62: Chef Support Updates and CI fixes
We need to drop chef 12 and 13 compatibility due to maintainability. See https://github.com/Netuitive/chef-netuitive/pull/61#issuecomment-504543833 for more details.

Make some decisions and document support, this includes outlining new branching strategies and expectations.

Fix ci

EDIT: based on the feedback here is a list of action items:
- [x] create branches for `chef-client/11.x`, `chef-client/12.x`, and `chef-client/13.x`.
- [x] These branches should be setup to be protected to prevent accidental deletion and such.
- [x] update in master readme documentation to help highlight these changes
- [x] remove chef 13 compatibility (depending on what version of 13.x you have you may already be broken but we can't control when upstream does breaking changes outside of major versions)
- [x] fixup CI

### PR DESCRIPTION
**Pull request by majormoses**
## Pull Request Checklist

closes #62 

#### General

- [ ] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [ ] Update README with any necessary configuration snippets

- [ ] RuboCop passes

- [ ] Existing tests pass

#### Purpose
Drop support for EOL versions, clarify support going forward for these. Fixup CI.

#### Known Compatibility Issues
Drops < chef 14 support, see `chef-client/$MAJOR.x` for previous chef version support.

### CODE REVIEW

**Comment by majormoses** [comment]
@ziggythehamster could use your :eyes: here as well.

File: README.md [hunk 1]
```diff
@@ -24,26 +24,44 @@ Supported Platforms
...
+
+The `master` branch is intended to be in line with the latest version of chef and its community based distributions. Whenever a breaking change is required to accomplish this we create a branch named `chef-client/$major_version.x`. Maintainers will only backport into the latest major version to match chefs [EOL Products](https://docs.chef.io/versions.html#end-of-life-eol-products) and their corresponding community distributions. Releases on Github and the Chef Supermarket will only be done from the master branches to support the latest versions of chef. EOL versions will continued to be versioned in their respective branches and can be easily pulled in via [berkshelf](https://docs.chef.io/berkshelf.html) which should support pulling from a branch or from a specific sha1.
+
+#### Chef 14.x
+
+This is currently supported by the master branch.
+
+#### Chef 13.x
+
+We supported chef 13 for quite a while after its EOL but its time has come. Please see `chef-client/13.x` branch for coninued usage and head branch for pull request.
```
Comment (rclagett) [inline README.md]: Small typo `coninued`

**Review by rclagett — APPROVED** [review]
LGTM

**Comment by majormoses** [comment]
released: https://supermarket.chef.io/cookbooks/netuitive/versions/1.0.0

**Pull request merged**